drupal/sites/default/boinc/modules/boincuser/boincuser.module

```diff
@@ -310,26 +352,6 @@ function boincuser_user($op, &$edit, &$account, $category = NULL) {
     case 'validate':
       if (isset($edit['validation_source'])) {
         switch ($edit['validation_source']) {
-        case 'user_register':
-          // Information being checked before adding a user
-          if (!boincuser_register_validate($edit)) {
-            // BOINC user validation failed for registration; set an error accordingly
-            form_set_error('mail', bts('An account already exists for @email. Log in or request password assistance to access your @project account.', array('@email' => $edit['mail'], '@project' => PROJECT), NULL, 'boinc:add-new-user'));
-          }
-          else {
-            // Save profile information for use during Insert
-            $_SESSION['profileInfo'] = array(
-              'country' => $edit['field_country'][0]['value'],
-              'zip' => $edit['field_zip'][0]['value'],                  
-              'url' => $edit['field_url'][0]['value'],
-              'background' => $edit['field_background'][0]['value'],
-              'opinions' => $edit['field_opinions'][0]['value']
-            );
-            // With BOINC validation passed, make sure name is unique
-            $edit['name'] = create_proper_drupalname($edit['boincuser_name']);
```

Comment:
This whole block of code may be obsolete now due to earlier changes in the registration process, but we should be sure the functionality of this particular line is handled elsewhere. The Drupal username is not exposed anywhere (ideally) to the end user, so the uniqueness of it should be handled automatically in the background by this create_proper_drupalname() function. (Drupal has a unique username requirement while BOINC does not, hence the need for this feature.)